auto_submit/test/validations/ci_successful_test.dart

```diff
@@ -2,45 +2,507 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'package:auto_submit/model/auto_submit_query_result.dart' hide PullRequest;
+import 'dart:convert';
+
+import 'package:github/github.dart' as github;
+import 'package:test/test.dart';
 import 'package:auto_submit/validations/ci_successful.dart';
+import 'package:auto_submit/model/auto_submit_query_result.dart';
 import 'package:auto_submit/validations/validation.dart';
-import 'package:github/github.dart';
-import 'package:test/test.dart';
 
-import '../requests/github_webhook_test_data.dart';
-import '../src/service/fake_config.dart';
-import '../src/service/fake_github_service.dart';
 import '../utilities/utils.dart';
 import '../utilities/mocks.dart';
+import '../src/service/fake_config.dart';
+import '../src/service/fake_github_service.dart';
+import '../src/service/fake_graphql_client.dart';
+import '../requests/github_webhook_test_data.dart';
 
 void main() {
-  late FakeConfig config;
   late CiSuccessful ciSuccessful;
-  late FakeGithubService githubService;
-  late RepositorySlug slug;
+  late FakeConfig config;
+  FakeGithubService githubService = FakeGithubService();
+  late FakeGraphQLClient githubGraphQLClient;
+  MockGitHub gitHub = MockGitHub();
+  late github.RepositorySlug slug;
+  late Set<FailureDetail> failures;
+
+  List<ContextNode> _getContextNodeListFromJson(final String repositoryStatuses) {
+    List<ContextNode> contextNodeList = [];
+
+    Map<String, dynamic> contextNodeMap = jsonDecode(repositoryStatuses) as Map<String, dynamic>;
+
+    dynamic statuses = contextNodeMap['statuses'];
```

Comment:
Nit: avoid dynamic.

Reply:
I normally try to but I can't here as the statuses are returned as a map since it is a nested object. So it is returned as Map<String, dynamic>.